packages/next-server/lib/router/router.js

```diff
@@ -224,6 +224,9 @@ export default class Router {
     }
 
     if (method !== 'pushState' || getURL() !== as) {
+    if (window.frameElement) {
```

Comment:
We don't do this check right 🤔

Reply:
So it probably should check for the method not being available

Reply:
@timneutkens The method IS available (I think), just restricted by the browser’s security model.

Reply:
@timneutkens Actually we don't need history to be rewritten if we run Next inside the iframe.